lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -1407,18 +1407,22 @@ add_node_scores_matching_attr(GHashTable *nodes, const pe_resource_t *rsc,
  * scores of the best nodes matching the attribute used for each of the
  * resource's relevant colocations.
  *
- * \param[in,out] rsc      Resource to check colocations for
- * \param[in]     log_id   Resource ID to use in logs (if NULL, use \p rsc ID)
- * \param[in,out] nodes    Nodes to update
- * \param[in]     attr     Colocation attribute (NULL to use default)
- * \param[in]     factor   Incorporate scores multiplied by this factor
- * \param[in]     flags    Bitmask of enum pcmk__coloc_select values
+ * \param[in,out] rsc         Resource to check colocations for
+ * \param[in]     log_id      Resource ID for logs (if NULL, use \p rsc ID)
+ * \param[in,out] nodes       Nodes to update
+ * \param[in]     colocation  Original colocation constraint (used to get
+ *                            configured primary resource's stickiness, and
+ *                            to get colocation node attribute; pass NULL to
+ *                            skip adding \p rsc's own node scores)
```

Comment:
For the shared `pcmk__add_colocated_node_scores()`, the actual effect is that we do nothing at all.`work` remains `NULL` if `colocation == NULL` and if we didn't call it via `cmp_resources()`.

Reply:
cmp_resources() is the only thing that passes NULL colocation (and that previously passed NULL attr). I should mention that NULL nodes, NULL colocation, and pcmk__coloc_select_this_with go together.

Reply:
Maybe log a message or something. `colocation == NULL` shouldn't happen unless `*nodes == NULL`.

Previously if `attr` wasn't set we pressed forward and defaulted it to `CRM_ATTR_UNAME` (as mentioned in another comment above). Now if `colocation == NULL` we'll fall through to `work == NULL` and bail out early.

Reply:
Added a CHECK to make sure everything is kosher